RocketChat/Rocket.Chat
# fix: Password Policy mismatch on user creation with random password

## Proposed changes (including videos or screenshots)


 - Use the `generate-password` library in order to create passwords that follow the provided password policy and avoid errors (**note:** errors can still happen when the "Forbid repeating characters" password policy is enabled, but these should be much less frequent);
 - **Hide** password text field when random password creation option is enabled on user creation. 



## Issue(s)

Fix #26726

## Steps to test or reproduce

Go to **Wokspace > Settings > Accounts > Password Policy** and enable it. Then go to **Users** and try creating a user with a random password. Errors can still happen on saving the user, but they should be less frequent.
When enabling the "Set random password and send by email" option, the password text field should be hidden:
![random-password](https://user-images.githubusercontent.com/36537004/228677427-52d4f036-7ff2-478f-9ceb-21ba57dcc415.gif)

## Further comments

[TC-193]


[TC-193]: https://rocketchat.atlassian.net/browse/TC-193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
